### PR TITLE
tests: add test for options parsing

### PR DESCRIPTION
**Pull request by qiweiii**
closes: #604

### CODE REVIEW

File: packages/chopsticks/src/schema/parse.test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+describe('Parsed options', () => {
+  const defaults = {
+    port: 8000,
+    'build-block-mode': 'Batch',
+  }
+  it('parsed multi type options should work', () => {
+    expect(
+      configSchema.parse({
+        block: 4500000,
```
Comment (xlc) [inline packages/chopsticks/src/schema/parse.test.ts:37]: add a test for `block: '4500000'`

File: packages/chopsticks/src/schema/parse.test.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  return arrayOfFiles
+}
+
+describe('Existing configs', async () => {
+  const configPaths = getAllFiles(path.join(__dirname, '../../../../configs'))
+  it.each(configPaths.map((p) => ({ name: _.last(p.split('/')), path: p })))(
+    '$name config parsing is correct',
+    async ({ path }) => {
+      const config = await fetchConfig(path)
+      expect(configSchema.parse(config)).toMatchSnapshot()
```
Comment (xlc) [inline packages/chopsticks/src/schema/parse.test.ts:24]: this is useful, with one issue that add or modify config will need to regenerate snapshots. maybe the easy way is just not check the output but only check parse is successful

**Comment by xlc** [comment]
I would like to test code in here as well https://github.com/AcalaNetwork/chopsticks/blob/3a8bcc792476c36448a3b28d3b8964f635bf80db/packages/core/src/setup.ts#L85-L104

**Comment by qiweiii** [comment]
> I would like to test code in here as well
> 
> https://github.com/AcalaNetwork/chopsticks/blob/3a8bcc792476c36448a3b28d3b8964f635bf80db/packages/core/src/setup.ts#L85-L104

agree, probably I shoud add an e2e test that read console output

**Review by xlc — APPROVED** [review]

**Review by xlc — APPROVED** [review]

**Pull request merged**